ansibleinviewer/main.py

```diff
@@ -40,12 +40,58 @@ def create_tmux_script(hosts: List[Host], vertical_panes) -> str:
     return tmux_script
 
 
+def slice_from_string(string: str):
+    """Parse standard indices string into slice type or integer
+    Slice is returned when range of indices is specified.
+    In case of single index, integer will be returned.
+
+    :param string: String to parse
+    :type string: str
+
+    :return: Slice or Integer with indices
+    :rtype: str or slice
+    """
+    if ':' in string:
+        return slice(*map(lambda x: int(x.strip()) if x.strip() else None, string.split(':')))
+    else:
+        if string.isdigit():
+            return int(string)
+        else:
+            print(f"Error parsing index {string}")
+            exit(1)
+
+
+def parse_inventory_groups(groups):
+    """Parse list of inventory groups passed via CLI
+    Groups with indices like: 3, [3:], [:3] should be parsed into slices
+    that later can be utilizes as list indices on inventory parsing
+
+    :param groups: List of strings with groups
+    :type groups: list
+
+    :return: List with tuples (name of group, slice of group)
+    :rtype: list
+    """
+
+    parsed_groups = []
+    for group in groups:
+        try:
+            open_bracket = group.index('[')
+            close_bracket = group.index(']')
+            parsed_group = (group[:open_bracket],
+                            slice_from_string(group[open_bracket+1:close_bracket]))
+        except ValueError:
+            parsed_group = (group, None)
+        parsed_groups.append(parsed_group)
+    return parsed_groups
+
```

Comment:
As mentioned above, I think user should be able to select only the whole group as he most probably won't know which slice of the group has the hosts he wants.

Reply:
In new patchset user can now select only whole groups and select groups that should be omitted